providers/default.rb

```diff
@@ -196,26 +196,70 @@ def run_post_install_action
         Chef::LocalMode.destroy_server_connectivity
       end
     end
-    Chef::Log.warn 'Replacing chef-client process with upgraded version and re-running.'
+    Chef::Log.warn "Replacing #{chef_infra_client_name} process with upgraded version and re-running."
     env = {}
     unless node['chef_client']['chef_license'].nil?
       env['CHEF_LICENSE'] = node['chef_client']['chef_license']
     end
     Kernel.exec(env, new_resource.exec_command, *new_resource.exec_args)
   when 'kill'
     if Chef::Config[:client_fork] && Process.ppid != 1 && !windows?
-      Chef::Log.warn 'Chef Infra Client is running forked with a supervisor. Sending KILL to parent process!'
+      Chef::Log.warn "#{chef_infra_product_name} is running forked with a supervisor. Sending KILL to parent process!"
       Process.kill('KILL', Process.ppid)
     end
-    Chef::Log.warn 'New Chef Infra Client installed and client process exit is allowed and/or specified. Now forcing Chef Infra Client to exit. Disregard any failure messages.'
+    Chef::Log.warn "New #{chef_infra_product_name} installed and client process exit is allowed and/or specified. Now forcing #{chef_infra_product_name} to exit. Disregard any failure messages."
     exit(213)
   else
     raise "Unexpected post_install_action behavior: #{new_resource.post_install_action}"
   end
 end
 
+def chef_infra_product_name
+  if defined?(::ChefUtils::Dist::Infra::PRODUCT)
+    ::ChefUtils::Dist::Infra::PRODUCT
+  else
+    'Chef Infra Client'
+  end
+end
+
+def chef_infra_short_name
+  if defined?(::ChefUtils::Dist::Infra::SHORT)
+    ::ChefUtils::Dist::Infra::SHORT
+  else
+    'chef'
+  end
+end
+
+def chef_infra_client_name
+  if defined?(::ChefUtils::Dist::Infra::CLIENT)
+    ::ChefUtils::Dist::Infra::CLIENT
+  else
+    'chef-client'
+  end
```

Comment:
If you care about old versions of cinc-client you may need to worry also about the dist constants before they were moved to `ChefUtils::`.  This looks appropriate for old-chef but i think there's old released versions of cinc-client that this will fail for.  I don't know how much that matters though.

Reply:
Hmm, that's a good point, hadn't considered that.

@ramereth What are your thoughts on this? Do we care about making this cookbook compatible with every released Cinc Client version? Or just all versions since `ChefUtils::` was introduced?

Reply:
@lamont-granquist I'm good with moving forward as-is. My changes add support for Cinc Client since `ChefUtils::`, which is 16.5.55+ I think. While it doesn't help with Cinc Clients before that it doesn't hurt either since they weren't working with this cookbook prior to these changes anyways.

Reply:
It looks like this was added in 16.5.64. Give 16.x is slated to be EOL "soon" I think this implementation is OK. We don't need to worry about that old of Cinc Clients for this.